src/coord/src/catalog.rs

```diff
@@ -1806,9 +1824,11 @@ impl Catalog {
         ) {
             // If an index exists for `id`, record it in the output set and stop
             // searching.
-            if let Some((index_id, _)) = catalog.indexes[&id].first() {
-                indexes.push(*index_id);
-                return;
+            if let Some(element) = catalog.indexes.get(&id) {
+                if let Some((index_id, _)) = element.first() {
+                    indexes.push(*index_id);
+                    return;
+                }
```

Comment:
Why did this change? The new code seems a little sketchy to me! There shouldn't ever be missing index IDs in the catalog!

Reply:
This changed because `uses` changed--instead of just returning dataflow dependencies, it's now returning all dependencies (including types). The other option is to create two functions (like `uses_dataflows` and `uses_other`), and then we won't need this change.

Reply:
Or I could do some filtering by `CatalogItem` type before calling `inner`!

Reply:
Ah, I see! Yeah, it seems weird to be recursively searching through types and functions that the view depends on. Filtering by `CatalogItem` type before calling `inner` sounds like the way to go to me.